Public/Src/Engine/Scheduler/Fingerprints/PipFingerprinter.cs

```diff
@@ -293,7 +295,7 @@ protected virtual void AddWeakFingerprint(IFingerprinter fingerprinter, Process
                 AddPipData(fingerprinter, "ResponseFileData", process.ResponseFileData);
             }
 
-            fingerprinter.AddCollection<EnvironmentVariable, ReadOnlyArray<EnvironmentVariable>>(
+            fingerprinter.AddOrderIndependentCollection<EnvironmentVariable, ReadOnlyArray<EnvironmentVariable>>(
                 "EnvironmentVariables",

```

Comment:
>"EnvironmentVariables", [](start = 16, length = 23)

Please add a unit test confirming that we still have a cache hit when environment variables are re-ordered.